File: leetCode-75/day-1/1480-running-sum-of-1d-array.js
/**
 * Runtime: 59ms
 * Memory Usage: 31.3 MB
 * Link: https://leetcode.com/submissions/detail/756400353/
 * //-----------------------------------------------------------------------------
 *    1. Loop over the nums
 *    2.    Add to nums[i] previous element
 *    3. return nums
 * //-----------------------------------------------------------------------------
 * @param {number[]} nums
 * @return {number[]}
 */
var runningSum = function(nums) {
    
    for(let i = 1; i < nums.length; i++) {

        nums[i]  += nums[i - 1];
    }

    return nums;
};

File: leetCode-75/day-1/1480/1480-running-sum-of-1d-array.cs
/**
 * Runtime: 148 ms
 * Memory Usage: 43.4 MB
 * Link: https://leetcode.com/submissions/detail/756419665/
//-----------------------------------------------------------------------------
   1. Loop over the nums
   2.    Add to nums[i] previous element
   3. return nums
//-----------------------------------------------------------------------------
 * @param {number[]} nums
 * @return {number[]}
 */
public class Solution {
    public int[] RunningSum(int[] nums) {

        for(int i = 1; i < nums.Length; i++)
        {
            nums[i] += nums[i - 1];
        }

        return nums;
    }
}